openmls/src/group/tests/test_commit_validation.rs

```diff
@@ -0,0 +1,841 @@
+            &[bob_group
+                .key_package_ref()
+                .expect("error retrieving kp ref")
+                .clone()],
+        )
+        .expect("Error creating remove")
+        .tls_serialize_detached()
+        .expect("Could not serialize message.");
+
+    let remove_message_in = erase_path(backend, &serialized_remove, &alice_group);
+
+    // Have Charlie process everything
+    let unverified_message = charlie_group
+        .parse_message(remove_message_in, backend)
+        .expect("Could not parse message.");
+
+    let err = charlie_group
+        .process_unverified_message(unverified_message, None, backend)
+        .expect_err("Could process unverified message despite missing path.");
+
+    assert_eq!(
+        err,
+        UnverifiedMessageError::InvalidCommit(StageCommitError::RequiredPathNotFound)
+    );
+
+    let original_remove_plaintext =
+        VerifiableMlsPlaintext::tls_deserialize(&mut serialized_remove.as_slice())
+            .expect("Could not deserialize message.");
+
+    // Positive case
+    let unverified_message = charlie_group
+        .parse_message(MlsMessageIn::from(original_remove_plaintext), backend)
+        .expect("Could not parse message.");
+    charlie_group
+        .process_unverified_message(unverified_message, None, backend)
+        .expect("Unexpected error.");
+}
+
+fn erase_path(
+    backend: &impl OpenMlsCryptoProvider,
+    mut pt: &[u8],
+    alice_group: &MlsGroup,
+) -> MlsMessageIn {
+    let mut plaintext =
+        VerifiableMlsPlaintext::tls_deserialize(&mut pt).expect("Could not deserialize message.");
+
+    // Keep the original plaintext for positive test later.
+    let original_plaintext = plaintext.clone();
+
+    let mut commit_content = if let MlsPlaintextContentType::Commit(commit) = plaintext.content() {
+        commit.clone()
+    } else {
+        panic!("Unexpected content type.");
+    };
+    commit_content.path = None;
+
+    plaintext.set_content(MlsPlaintextContentType::Commit(commit_content));
+
+    let alice_credential_bundle = backend
+        .key_store()
+        .read(
+            &alice_group
+                .credential()
+                .expect("error retrieving credential")
+                .signature_key()
+                .tls_serialize_detached()
+                .expect("error serializing credential"),
+        )
+        .expect("error retrieving credential bundle");
+
+    let serialized_context = alice_group
+        .export_group_context()
+        .tls_serialize_detached()
+        .expect("error serializing context");
+    plaintext.set_context(serialized_context.clone());
+
+    // We have to re-sign, since we changed the content.
+    let mut signed_plaintext: MlsPlaintext = plaintext
+        .payload()
+        .clone()
+        .sign(backend, &alice_credential_bundle)
+        .expect("Error signing modified payload.");
+
+    // Set old confirmation tag
+    signed_plaintext.set_confirmation_tag(
```

Comment:
Is it really necessary the set the confirmation tag and membership tag? (this is also done elsewhere)

Reply:
When re-signing the plaintext, both confirmation and membership tag are set to `None`. I guess that is at least partially due to limitations of the trait and in general it probably makes sense. Although I suppose it would be better to have setting and verification of the confirmation and membership tags encoded as type transitions in the same way as with signatures. I think there's an issue for that somewhere, but we haven't come around to it.